go/fixchain/url_cache.go

```diff
@@ -20,7 +43,7 @@ type urlCache struct {
 }
 
 func (u *urlCache) getURL(url string) ([]byte, error) {
-	r, ok := u.cache[url]
+	r, ok := u.cache.get(url)
```

Comment:
You do need locking to be thread safe, as multiple goroutines will be potentially reading from and writing to this one cache at the same time.  My tests went into a frenzy of panicking goroutines once or twice due to this :)